### PR TITLE
[10.x] Re-formatted CI and added more PHP versions to the test matrix.

### PR DESCRIPTION
**Pull request by AlexSkrypnyk**
I've updated the CI config to have names for each job for consistency.

I've also added additional PHP versions to test:
| PHP Versions | Drupal Release | Composer Channel |
|--------------|----------------|------------------|
| 8.1          | stable         | stable           |
| 8.1          | stable         | snapshot         |
| 8.1          | dev            | stable           |
| 8.1          | dev            | snapshot         |
| 8.2          | stable         | stable           |
| 8.2          | stable         | snapshot         |
| 8.2          | dev            | stable           |
| 8.2          | dev            | snapshot         |
| 8.3          | stable         | stable           |
| 8.3          | stable         | snapshot         |

Note that Drupal 10.x-dev tests fail on PHP 8.3. https://www.drupal.org/project/drupal/issues/3375693

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -53,24 +60,25 @@ jobs:
...
-        # https://www.drupal.org/project/drupal/issues/3182653
-        # https://www.drupal.org/node/3176567
-      - if: ${{ matrix.php-versions == '8.1' }}
-        run: composer require phpspec/prophecy-phpunit:^2
-
-      - name: Run a single unit test to verify the testing setup.
+      - name: Run a single unit test to verify the testing setup
         run: ./vendor/bin/phpunit -c ./web/core ./web/core/modules/user/tests/src/Unit/UserAccessControlHandlerTest.php
-
-      - run: ./vendor/bin/drush
```
Comment (AlexSkrypnyk) [inline .github/workflows/ci.yml:76]: this is not needed - Drush was used above to install the site and run the server

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -53,24 +60,25 @@ jobs:
...
 
-      - run: until curl -s $SIMPLETEST_BASE_URL; do true; done > /dev/null
+      - name: Start server
+        run: |
+          ./vendor/bin/drush runserver "$SIMPLETEST_BASE_URL" &
+          until curl -s "$SIMPLETEST_BASE_URL"; do true; done > /dev/null
 
-        # https://www.drupal.org/project/drupal/issues/3182653
-        # https://www.drupal.org/node/3176567
-      - if: ${{ matrix.php-versions == '8.1' }}
```
Comment (AlexSkrypnyk) [inline .github/workflows/ci.yml:70]: moved above to the part where all of the code base is prepared

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -53,24 +60,25 @@ jobs:
...
-
-      - if: matrix.drupal-release == 'dev'
-        run: composer --verbose update
+      # @see https://www.drupal.org/node/3176567
+      - name: Install testing dependency
+        if: ${{ matrix.php-versions == '8.1' }}
+        run: composer require --dev phpspec/prophecy-phpunit:^2
 
-      - run: ./vendor/bin/drush site-install --verbose --yes --db-url=sqlite://tmp/site.sqlite
+      - name: Override Drupal version to dev for testing dev releases
```
Comment (AlexSkrypnyk) [inline .github/workflows/ci.yml:68]: this is a single operation, so combined multiple steps

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -15,17 +18,21 @@ jobs:
     runs-on: 'ubuntu-20.04'
     strategy:
       fail-fast: false
+
       matrix:
-        php-versions: ['8.1']
-        drupal-release: ['stable']
-        composer-channel: ['stable']
+        php-versions: ['8.1', '8.2']
```
Comment (AlexSkrypnyk) [inline .github/workflows/ci.yml:23]: | PHP Versions | Drupal Release | Composer Channel |
|--------------|----------------|------------------|
| 8.1          | stable         | stable           |
| 8.1          | stable         | snapshot         |
| 8.1          | dev            | stable           |
| 8.1          | dev            | snapshot         |
| 8.2          | stable         | stable           |
| 8.2          | stable         | snapshot         |
| 8.2          | dev            | stable           |
| 8.2          | dev            | snapshot         |
| 8.3          | stable         | stable           |
| 8.3          | stable         | snapshot         |

**Comment by AlexSkrypnyk** [comment]
@leymannx 
I will create similar D11 PR once I get a review/feedback on this one

**Review by normanlolx — APPROVED** [review]
This looks very good to me 👌🏻

**Pull request merged**